taichi/aot/module_loader.cpp

```diff
@@ -4,15 +4,20 @@ namespace taichi {
 namespace lang {
 namespace aot {
 
-Kernel *ModuleLoader::get_kernel(const std::string &name) {
-  auto itr = loaded_kernels_.find(name);
-  if (itr != loaded_kernels_.end()) {
-    return itr->second.get();
+std::unique_ptr<Module> Module::load(const std::string &path,
+                                     Arch arch,
+                                     std::any mod_params) {
+  if (arch == Arch::vulkan) {
+#ifdef TI_WITH_VULKAN
```

Comment:
Again, you don't have to macro-protect here. `vulkan::make_aot_module` itself can use a `TI_WITH_VULKAN` check

Reply:
https://github.com/taichi-dev/taichi/issues/4464